### PR TITLE
Feature / Added ability for custom fields to show on form.

### PR DESCRIPTION
**Pull request by thekevinbrown**
We had custom fields already in the table, but not in the detail form.

This PR extends this ability so that custom fields can show in both the table, and the form.

**Breaking Change**: The arguments for custom fields have changed. Instead of just being `entity`, now it's `{ entity, context }` where context tells you the context you're rendering in.

If users specify `showOn` then the field only shows in the places they specify, while if they do not, it shows everywhere.

Users can also override default fields with custom fields by giving them the same name. At that point the custom field wins.

I also refactored the field components into separate files while I was in there, but this was not strictly related to this change.

### CODE REVIEW

File: src/packages/admin-ui-components/src/utils/use-schema.ts [hunk 1]
```diff
@@ -32,13 +32,7 @@ export enum AdminUIFilterType {
...
 }
 
-export enum EntityFieldType {
-	BOOLEAN = 'Boolean',
-	ID = 'ID!',
-	OPTIONAL_ID = 'ID',
-	JSON = 'JSON',
-	// TODO: This needs to be extended for the other types
-}
+export type EntityFieldType = 'Boolean' | 'custom' | 'ID!' | 'ID' | 'JSON';
```
Comment (steve-keep) [inline src/packages/admin-ui-components/src/utils/use-schema.ts:35]: Curious why you removed the enum?

File: src/examples/rest/src/admin-ui/custom-fields/link.tsx [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+		value: string;
+	};
+}
+
+export const Link = ({ entity }: CustomFieldArgs<Task>) => {
 	const handleClick = (e: MouseEventHandler<HTMLDivElement>) => {
 		e.preventDefault();
 		e.stopPropagation();
-		window.open(`https://google.com/search?q=${task.user.name}`, '_blank', 'noreferrer');
+		window.open(`https://google.com/search?q=${entity.user.label}`, '_blank', 'noreferrer');
```
Comment (steve-keep) [inline src/examples/rest/src/admin-ui/custom-fields/link.tsx:17]: Does the component get all the data on that row?

File: src/packages/admin-ui-components/src/utils/use-schema.ts (hunk 1 above)
Comment (thekevinbrown) [reply src/packages/admin-ui-components/src/utils/use-schema.ts:35]: It wouldn't work from the user's custom fields definition. The import was undefined, so I just went to strings.

File: src/examples/rest/src/admin-ui/custom-fields/link.tsx (hunk 2 above)
Comment (thekevinbrown) [reply src/examples/rest/src/admin-ui/custom-fields/link.tsx:17]: Yup. `entity` is the row object.

File: src/packages/admin-ui-components/src/detail-panel/component.tsx [hunk 3]
```diff
@@ -181,6 +62,17 @@ const DetailField = ({ field }: { field: EntityField }) => {
 	);
 };
 
+const CustomField = ({ field }: { field: CustomField }) => {
```
Comment (steve-keep) [inline src/packages/admin-ui-components/src/detail-panel/component.tsx:65]: Should this also be in the fields folder?

**Review by steve-keep — APPROVED** [review]
Looks good, just one small comment about moving the custom field into a file with the other fields.

File: src/packages/admin-ui-components/src/detail-panel/component.tsx (hunk 3 above)
Comment (thekevinbrown) [reply src/packages/admin-ui-components/src/detail-panel/component.tsx:65]: On this one, it's actually an alternative to `DetailField`, which is also in this file, e.g. the things in the folder are the actual fields, where these can more be called "wrappers"... What do you reckon?

Comment (steve-keep) [reply src/packages/admin-ui-components/src/detail-panel/component.tsx:65]: sounds good 👍

**Pull request merged**